operate/client/src/modules/stores/processInstances.ts

```diff
@@ -176,6 +176,29 @@ class ProcessInstances extends NetworkReconnectionHandler {
     }
   };
 
+  getProcessInstanceOperationError = (
+    selectedId: ProcessInstanceEntity['id'],
+  ) => {
+    const instance = this.state.processInstances.find(({id, operations}) => {
+      const isMatchedId = id === selectedId;
+      const hasFailedOperation = operations.find(
+        ({state}) => state === 'FAILED',
+      );
+
+      return isMatchedId && hasFailedOperation;
+    });
+    const {batchOperationId} = getProcessInstancesRequestFilters();
+    const hasOperationIdFilterActive = instance?.operations.find(
+      ({batchOperationId: batchId}) => batchId === batchOperationId,
+    );
+
+    if (!!hasOperationIdFilterActive) {
+      return instance?.operations[0]?.errorMessage ?? null;
+    }
```

Comment:
Same thing I mentioned earlier: `instance?.operations[0]` might return a wrong operation.

Plus I think there is potential to shorten the whole function, because `hasOperationIdFilterActive` should get you the correct operation already.